src/sql/base/browser/ui/table/plugins/rowNumberColumn.plugin.ts

```diff
@@ -29,7 +33,7 @@ export class RowNumberColumn<T> implements Slick.Plugin<T> {
 	}
 
 	private handleClick(e: MouseEvent, args: Slick.OnClickEventArgs<T>): void {
-		if (this.grid.getColumns()[args.cell].id === 'rowNumber') {
+		if (this.grid.getColumns()[args.cell].id === 'rowNumber' && this.options.autoCellSelection !== false) {
```

Comment:
Why not this.options.autoCellSelection

Reply:
the default value for this option is true

Reply:
I believe he's asking why the check isn't `(this.grid.getColumns()[args.cell].id === 'rowNumber' && this.options.autoCellSelection)`

Which the reason is that the "default" is supposed to be true, but if no options are given then options defaults to an empty object. 

@alanrenmsft I think it'd be clearer to make a default options object and then merge the two in the constructor so you're sure that it actually has a value (and the value is true) if nothing is specified.

Reply:
yes, that's what I am working on. mixin the options with defaults.